placeholder/src/main/java/com/google/accompanist/placeholder/PlaceholderAnimatedBrush.kt

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.placeholder
+
+import androidx.compose.animation.core.AnimationSpec
+import androidx.compose.animation.core.InfiniteRepeatableSpec
+import androidx.compose.animation.core.LinearEasing
+import androidx.compose.animation.core.RepeatMode
+import androidx.compose.animation.core.infiniteRepeatable
+import androidx.compose.animation.core.tween
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.graphics.lerp
+
+internal object PlaceholderDefaults {
+    val PlaceholderColor = Color.Gray.copy(alpha = 0.5f)
+    val PlaceholderHighlightColor = Color.Gray.copy(alpha = 0.3f)
+}
+
+/**
+ * Creates a [Fade] brush with the given initial and target colors.
+ *
+ * @sample com.google.accompanist.sample.placeholder.PlaceholderFadeSample
+ *
+ * @param initialColor initial color of the fade.
+ * @param targetColor target color of the fade.
+ * @param animationSpec the [AnimationSpec] to configure the animation.
+ */
+fun fadeBrush(
```

Comment:
Could this be an ext-fun on `PlaceholderAnimatedBrush` instead? `PlaceholderAnimatedBrush.fade(...)`

Reply:
🙆‍♂️

Reply:
Sorry, I didn't fully understand your suggestion.
Are you talking about moving inside the `companion object`?

Reply:
Sorry yes.